[Chetwin Low] iP

chetwinlow opened the pull request:
# DukeBro

> "DUKE THOOOOOO" - Anon

When your memory is bad and you need a friend to talk to, DukeBro is here for you, it's:

* text-based
* really easy to learn
* SUPER RESPONSIVE
* All you gotta to do is,

1. Download it from [here](https://github.com/chetwinlow/ip/releases/tag/A-Jar)
2. Use your terminal to run it duhhh
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FOC**!

Features:

- [x] Managing tasks
- [ ] Being your best friend (coming soon this year)
- [ ] Taking over the world (coming soon in 2030)

Please marvel at the following code block that demonstrates my coding superiority, here is `main`:

```javascript
public class Main {
    public static void main(String[] args) {
        System.out.print("Have a nice day! Take a break sometimes.");
    }
}
```

Commit:
Add Gradle support

chetwinlow committed:
Level-1 Completed

chetwinlow committed:
Level-2 Completed

chetwinlow committed:
Level-3 Completed

chetwinlow committed:
Level-4 completed

chetwinlow committed:
A-TextUiTesting completed

chetwinlow committed:
Level-5 completed

chetwinlow committed:
Level-6 completed

chetwinlow committed:
Level-7 Completed

chetwinlow committed:
Level-8 Completed

chetwinlow committed:
Merge pull request #1 from chetwinlow/branch-Level-7

Level-7 Completed

chetwinlow committed:
Merge pull request #2 from chetwinlow/branch-Level-8

Level-8 Completed

chetwinlow committed:
A-MoreOOP Completed

chetwinlow committed:
A-Packages Completed

chetwinlow committed:
A-JUnit Completed

chetwinlow committed:
A-JavaDoc Completed

chetwinlow committed:
A-CodingStandard completed

chetwinlow committed:
Level-9 completed

chetwinlow committed:
Merge pull request #3 from chetwinlow/branch-A-JavaDoc

A-JavaDoc Completed

chetwinlow committed:
Merge pull request #4 from chetwinlow/branch-A-CodingStandard

A-CodingStandard completed

chetwinlow committed:
Merge pull request #5 from chetwinlow/branch-Level-9

Level-9 completed

chetwinlow committed:
Minor changes to Add.

chetwinlow committed:
Merge pull request #6 from chetwinlow/add-gradle-support

Add Gradle support

jhlee1997 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+     *
+     * @return Whether this is an Exit.
+     */
+    public boolean isExit() {
```
Since this method is setting the boolean variable ```canExit``` into ```false```, perhaps it is better to name this as ```seteCanExit()```
```suggestion
    public boolean setCanExit() {
```

jhlee1997 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,108 @@
+    /**
+     * Converts a Task into a string.
+     *
+     * @param task THe given Task.
```
Just a minor spelling mistake!
```suggestion
     * @param task The given Task.
```

jhlee1997 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,108 @@
+     * @param fullString The given string
+     * @return The Task with the appropriate information.
+     */
+    public static Task stringToTask(String fullString) {
```
Perhaps it is better to name this method ```convertToTask()``` such that the method name starts with a verb
```suggestion
    public static Task convertToTask(String fullString) {
```

gordonlzy replied:
I agree with jhlee1997 that convertToTask might be better as a method name

jhlee1997 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
Perhaps it is better name this boolean variable as ```canExit```
```suggestion
        boolean canExit = false;
```

jhlee1997 reviewed (commented):
Hi Chetwin, I am Jae and this is my first PR review about coding standard violations.
Overall, I really appreciate the clean coding style of yours!
Just picked out 4 minor adjustments to make. Thanks!

gordonlzy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+ * Represents a wrapper class for an arraylist of Tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> list;
```
Would it be better to use tasks instead of list to show that it is a collection of tasks and it is in plural form?

gordonlzy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,41 @@
+     * @throws EmptyDescriptionException If the string contains todo but no following description.
+     */
+    public static Command parse(String input) throws IllegalCommandException, EmptyDescriptionException {
+        if (input.equals("bye")) {
```
Do you think that the usage of switch here will work better?

gordonlzy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,41 @@
+package duke;
+
+import duke.command.*;
```
Perhaps the imported classes should be listed explicitly?

gordonlzy reviewed (commented):
LGTM, just a few nitpicks

chetwinlow committed:
Level-10 completed

chetwinlow committed:
Merge pull request #7 from chetwinlow/branch-Level-10

Level-10 completed

chetwinlow committed:
Level-10 completed

chetwinlow committed:
Merge pull request #8 from chetwinlow/branch-Level-10

Level-10 completed

chetwinlow committed:
A-Gradle completed

chetwinlow committed:
Merge pull request #9 from chetwinlow/add-gradle-support

A-Gradle completed

chetwinlow committed:
A-CheckStyle completed

chetwinlow committed:
Merge branch 'master' into branch-A-CheckStyle

chetwinlow committed:
Merge pull request #10 from chetwinlow/branch-A-CheckStyle

A-CheckStyle completed

chetwinlow committed:
Assertions added.

Documents important assumptions that should hold at various
points of the code.

Assertions ensure the correctness of any assumptions
which have been done in the program.

Assert statements are used before calls to methods.

To ensure arguments to methods are valid.

chetwinlow committed:
Improvements in code quality

Some method names do not abide to code standards.

Poor code standards may cause confusion to readers
or collaborators.

Reviewing and make appropriate changes.

There will be less confusion for readers.

chetwinlow committed:
Merge pull request #11 from chetwinlow/A-Assertions

Assertions added.

chetwinlow committed:
Merge pull request #12 from chetwinlow/A-CodeQuality

Improvements in code quality

chetwinlow committed:
Priority added.

Tasks have no indication of priority.

Users can see how urgent or important each task is.

A letter L=Low, M=Medium, H=High to indicate priority level.

Clear and concise.

chetwinlow committed:
Merge pull request #13 from chetwinlow/C-Priority

Priority added.

chetwinlow committed:
Set theme jekyll-theme-merlot

chetwinlow committed:
User Guide added.

chetwinlow committed:
Merge remote-tracking branch 'origin/master'

chetwinlow committed:
User Guide added.

chetwinlow committed:
Final changes

chetwinlow committed:
Update User Guide

chetwinlow committed:
Updated User Guide styling.
